lib/gui/menu.js

```diff
@@ -18,99 +18,139 @@
 
 const electron = require('electron')
 
-const menuTemplate = [
-  {
-    role: 'editMenu'
-  },
-  {
-    label: 'View',
-    submenu: [
-      {
-        role: 'toggledevtools',
-        click () {
-          const window = electron.BrowserWindow.getFocusedWindow()
-          if (window) {
-            window.webContents.openDevTools({
-              mode: 'detach'
-            })
+/* eslint-disable no-magic-numbers */
+
+/**
+ * @summary Builds a native application menu for a given window
+ * @param {electron.BrowserWindow} window - BrowserWindow instance
+ * @example
+ * buildWindowMenu(mainWindow)
+ */
+const buildWindowMenu = (window) => {
+  /**
+   * @summary Toggle the main window's devtools
+   * @example
+   * toggleDevTools()
+   */
+  const toggleDevTools = () => {
+    if (!window) {
+      return
+    }
+
+    // NOTE: We can't use `webContents.toggleDevTools()` here,
+    // as we need to force detached mode
+    if (window.webContents.isDevToolsOpened()) {
+      window.webContents.closeDevTools()
+    } else {
+      window.webContents.openDevTools({
+        mode: 'detach'
+      })
+    }
+  }
+
+  /**
+   * @summary Open the main window's settings page
+   * @example
+   * showSettings()
+   */
+  const showSettings = () => {
+    if (window) {
+      window.webContents.send('menu:preferences')
+    }
+  }
+
+  const menuTemplate = [
+    {
+      role: 'editMenu'
+    },
+    {
+      role: 'windowMenu'
+    },
+    {
+      role: 'help',
+      submenu: [
+        {
+          label: 'Etcher Pro',
+          click () {
+            electron.shell.openExternal('https://etcher.io/pro')
+          }
+        },
+        {
+          label: 'Etcher Website',
+          click () {
+            electron.shell.openExternal('https://etcher.io')
+          }
+        },
+        {
+          label: 'Report an issue',
+          click () {
+            electron.shell.openExternal('https://github.com/resin-io/etcher/issues')
           }
         }
-      }
-    ]
-  },
-  {
-    role: 'windowMenu'
-  },
-  {
-    role: 'help',
-    submenu: [
-      {
-        label: 'Etcher Pro',
-        click () {
-          electron.shell.openExternal('https://etcher.io/pro')
-        }
-      },
-      {
-        label: 'Etcher Website',
-        click () {
-          electron.shell.openExternal('https://etcher.io')
+      ]
+    }
+  ]
+
+  if (process.platform === 'darwin') {
+    menuTemplate.unshift({
+      label: electron.app.getName(),
+      submenu: [ {
+        role: 'about'
+      }, {
+        type: 'separator'
+      }, {
+        label: 'Preferences',
+        accelerator: 'Command+,',
+        click: showSettings
+      }, {
+        type: 'separator'
+      }, {
+        role: 'hide'
+      }, {
+        role: 'hideothers'
+      }, {
+        role: 'unhide'
+      }, {
+        type: 'separator'
+      }, {
+        role: 'quit'
+      } ]
+    })
+    menuTemplate.splice(2, 0, {
+      label: 'View',
+      submenu: [
```

Comment:
This same thing is added regardless of the platform. Can we declare it once? You can fix the `Command` and `Ctrl` differences by using an accelerator that join both (`CtrlOrCmd` or something like that) as described in the docs

Reply:
Good point, the accelerator is completely different though; on Linux / Windows it's using the <kbd>Shift</kbd> modifier, whereas on Mac OS it's <kbd>Alt</kbd>

Reply:
See also #2001 :wink:

Reply:
@jviotti de-duplicated that submenu now